lib/libutee/tee_api_arith_mpi.c

```diff
@@ -785,28 +821,56 @@ int32_t TEE_BigIntIsProbablePrime(const TEE_BigInt *op,
  * own. Performance of RSA operations in TEE Internal API are not affected
  * by this.
  */
-void TEE_BigIntInitFMM(TEE_BigIntFMM *bigIntFMM, uint32_t len)
+void TEE_BigIntInitFMM(TEE_BigIntFMM *bigIntFMM, size_t len)
 {
 	TEE_BigIntInit(bigIntFMM, len);
 }
 
+void __GP11_TEE_BigIntInitFMM(TEE_BigIntFMM *bigIntFMM, uint32_t len)
+{
+	TEE_BigIntInitFMM(bigIntFMM, len);
+}
+
 void TEE_BigIntInitFMMContext(TEE_BigIntFMMContext *context __unused,
-			      uint32_t len __unused,
+			      size_t len __unused,
 			      const TEE_BigInt *modulus __unused)
 {
 }
 
-uint32_t TEE_BigIntFMMSizeInU32(uint32_t modulusSizeInBits)
+void __GP11_TEE_BigIntInitFMMContext(TEE_BigIntFMMContext *context,
+				     uint32_t len, const TEE_BigInt *modulus)
+{
+	TEE_BigIntInitFMMContext(context, len, modulus);
+}
+
+TEE_Result TEE_BigIntInitFMMContext1(TEE_BigIntFMMContext *context __unused,
+				     size_t len __unused,
+				     const TEE_BigInt *modulus __unused)
+{
+	return TEE_SUCCESS;
```

Comment:
("GP131: Update TEE_BigInt*() functions") `TEE_ERROR_NOT_IMPLEMENTED`?

Reply:
No, this is intended behaviour.

Reply:
It's an opportunity to do fast modular multiplication, but we've never bothered to do the fast part of FMM so we just use the regular bigint function. There is a comment a bit further up in this file that explains a bit. I'd also recommend looking in the specification.

Reply:
Ok, right. thanks. 
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commit "GP131: Update TEE_BigInt*() functions".